packages/dds/tree/src/feature-libraries/modular-schema/discrepancies.ts

```diff
@@ -182,7 +180,7 @@ export function getAllowedContentIncompatibilities(
 					stored: undefined,
 				} satisfies NodeKindIncompatibility);
 			} else {
-				const storedNodeSchema = stored.nodeSchema.get(key);
+				const storedNodeSchema = stored.nodeSchema.get(key) as TreeNodeStoredSchema;
```

Comment:
nit: non-null assertion (valid because you used a has check above) is a bit better. you could avoid it entirely by using .get unconditionally and checking if the return is undefined, which is also slightly more efficient and what i'd tend to do.